packages/app-check/lib/index.d.ts

```diff
@@ -259,7 +261,7 @@ export namespace FirebaseAppCheckTypes {
      * @returns A function that unsubscribes this listener.
      */
     // TODO there is a great deal of Observer / PartialObserver typing to carry-in
-    // onTokenChanged(observer: PartialObserver<AppCheckTokenResult>): () => void;
+    onTokenChanged(observer: AppCheckListenerCallback): () => void;
```

Comment:
🤔 we must follow the upstream typings here, they are our source of truth for typing reference

https://github.com/firebase/firebase-js-sdk/blob/25cda8af6a6fc15e33f0ce5644dd29d996e4716f/packages/app-check-types/index.d.ts#L69

```typescript
   onTokenChanged(observer: PartialObserver<AppCheckTokenResult>): Unsubscribe;
```

Reply:
Ok I'll update code to match the js sdk.